pkg/collector/corechecks/cluster/kubernetes_state_aggregators.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+// +build kubeapiserver
+
+package cluster
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	ksmstore "github.com/DataDog/datadog-agent/pkg/kubestatemetrics/store"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type metricAggregator interface {
+	accumulate(ksmstore.DDMetric)
+	flush(aggregator.Sender, *KSMCheck, *labelJoiner)
+}
+
+// maxNumberOfAllowedLabels contains the maximum number of labels that can be used to aggregate metrics.
+// The only reason why there is a maximum is because the `accumulator` map is indexed on the label values
+// and GO accepts arrays as valid map key type, but not slices.
+// This hard coded limit is fine because the metrics to aggregate and the label list to use are hardcoded
+// in the code and cannot be arbitrarily set by the end-user.
+const maxNumberOfAllowedLabels = 3
+
+type counterAggregator struct {
+	metricName    string
+	allowedLabels []string
+
+	accumulator map[[maxNumberOfAllowedLabels]string]float64
+}
+
+func newCounterAggregator(metricName string, allowedLabels []string) metricAggregator {
+	if len(allowedLabels) > maxNumberOfAllowedLabels {
+		log.Error("BUG in KSM metric aggregator")
```

Comment:
mind elaborating the error message? 🙇

Reply:
I just added a comment just above the test to explain why the condition cannot be true unless there’s a bug in the code.